pyproject.toml

```diff
@@ -1,80 +1,63 @@
+repository = "https://github.com/Codium-ai/pr-agent"
+authors = ["Itamar Friedman <itamar.f@codium.ai>"]
 maintainers = [
-  {name = "Ori Kotek", email = "ori.k@codium.ai"},
-  {name = "Tal Ridnik", email = "tal.r@codium.ai"},
-  {name = "Hussam Lawen", email = "hussam.l@codium.ai"},
-  {name = "Sagi Medina", email = "sagi.m@codium.ai"}
+    "Ori Kotek <ori.k@codium.ai>",
+    "Tal Ridnik <tal.r@codium.ai>",
+    "Hussam Lawen <hussam.l@codium.ai>",
+    "Sagi Medina <sagi.m@codium.ai>",
 ]
-description = "CodiumAI PR-Agent is an open-source tool to automatically analyze a pull request and provide several types of feedback"
-readme = "README.md"
-requires-python = ">=3.9"
 keywords = ["ai", "tool", "developer", "review", "agent"]
-license = {file = "LICENSE", name = "Apache 2.0 License"}
+license = "Apache 2.0 License"
+readme = "README.md"
 classifiers = [
   "Development Status :: 3 - Alpha",
   "Intended Audience :: Developers",
   "Operating System :: Independent",
   "Programming Language :: Python :: 3",
 ]
-dynamic = ["dependencies"]
-
-[tool.setuptools.dynamic]
-dependencies = {file = ["requirements.txt"]}
-
-[project.urls]
-"Homepage" = "https://github.com/Codium-ai/pr-agent"
-
-[tool.setuptools]
-include-package-data = false
-license-files = ["LICENSE"]
-
-[tool.setuptools.packages.find]
-where = ["."]
-include = ["pr_agent"]
-
-[project.scripts]
-pr-agent = "pr_agent.cli:run"
-
-
-[tool.ruff]
-
-line-length = 120
-
-select = [
-  "E",  # Pyflakes
-  "F",  # Pyflakes
-  "B",  # flake8-bugbear
-  "I001",  # isort basic checks
-  "I002",  # isort missing-required-import
-  ]
-
-# First commit - only fixing isort
-fixable = [
-  "I001",  # isort basic checks
-]
-
-unfixable = [
-  "B",  # Avoid trying to fix flake8-bugbear (`B`) violations.
-  ]
-
-exclude = [
-  "api/code_completions",
-]
+include = ["CHANGELOG.md"]
+
+[tool.poetry.dependencies]
+python = "^3.10"
+aiohttp = "3.9.1"
+atlassian-python-api = "3.39.0"
+azure-devops = "7.1.0b3"
+boto3 = "1.33.6"
+dynaconf = "3.2.4"
+fastapi = "0.99.0"
+gitpython = "3.1.32"
+google-cloud-aiplatform = "1.35.0"
+google-cloud-storage = "2.10.0"
+jinja2 = "3.1.2"
+litellm = "0.12.5"
+loguru = "0.7.2"
+msrest = "0.7.1"
+openai = "0.27.8"
+pygithub = "==1.59.*"
+pyyaml = "6.0.1"
+python-gitlab = "3.15.0"
+retry = "0.9.2"
+starlette-context = "0.3.6"
+tiktoken = "0.5.2"
+ujson = "5.8.0"
+uvicorn = "0.22.0"
+pinecone-client = "^2.2.4"
+pinecone-datasets = {git = "https://github.com/mrT23/pinecone-datasets.git", rev = "main"}
+
+
+[tool.poetry.group.dev.dependencies]
+pytest = "^7.4.3"
+httpx = "^0.26.0"
+ruff = "^0.1.9"
 
-ignore = [
-  "E999", "B008"
-]
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
 
-[tool.ruff.per-file-ignores]
-"__init__.py" = ["E402"]  # Ignore `E402` (import violations) in all `__init__.py` files, and in `path/to/file.py`.
-# TODO: should decide if maybe not to ignore these.
+[tool.poetry.scripts]
+pr-agent = "python pr_agent.cli:run"
```

Comment:
/ask explain these changes

Reply:
The selected lines do not show any specific changes within the `pyproject.toml` file. This indicates that the context provided does not include any additions, deletions, or modifications. 

However, based on the PR title "Introduce Poetry as Dependency Manager and Update Dockerfiles," it is likely that the `pyproject.toml` file has been added or modified to include dependency management configurations for Poetry. This file typically contains metadata about the project, including dependencies, which Poetry uses to manage the project's environment. 

To provide a more detailed explanation, specific lines showing the actual changes in the `pyproject.toml` file would be necessary.

Reply:
The selected line:

```diff
+pr-agent = "python pr_agent.cli:run"
```

is part of the `[tool.poetry.scripts]` section in the `pyproject.toml` file. This line defines a command-line script entry point for the project when using Poetry as the dependency manager.

Explanation:
- `pr-agent` is the name of the command that will be available to the user after installing the package.
- `"python pr_agent.cli:run"` specifies the Python command that should be executed when the user runs `pr-agent`. Here, it indicates that the `run` function from the `pr_agent.cli` module should be executed.

This change allows users to run the `pr-agent` command directly from the command line, simplifying the execution of the tool.

Reply:
The selected lines do not show any specific changes within the `pyproject.toml` file. The context hunk provided is empty, which means no additions, deletions, or unchanged lines are displayed.

However, based on the PR title "Introduce Poetry as Dependency Manager and Update Dockerfiles," we can infer that the `pyproject.toml` file likely includes changes related to configuring Poetry as the dependency manager. Typically, this would involve:

- Adding project metadata (e.g., name, version, description).
- Specifying dependencies and their versions.
- Including build system requirements.

To provide a more detailed explanation, specific lines from the `pyproject.toml` file would be necessary.

Reply:
The selected lines do not show any specific changes from the `pyproject.toml` file. This means that the provided context does not include any additions, deletions, or modifications.

However, based on the PR title "Introduce Poetry as Dependency Manager and Update Dockerfiles," it is likely that the `pyproject.toml` file has been added or modified to include dependencies and configurations managed by Poetry. This file typically includes sections such as `[tool.poetry]`, `[tool.poetry.dependencies]`, and `[tool.poetry.dev-dependencies]`, which define the project's metadata and dependencies.

To provide a more detailed explanation, specific lines from the `pyproject.toml` file showing the actual changes would be needed.